### ISSUE TITLE
Sample API returns only a single URL even when multiple samples are uploaded in a zip file

### ISSUE THREAD

**Issue by Lambdanaut** [crits/crits#670]
When a zip(or rar) archive with multiple files inside is uploaded over the Samples API, a message is returned along with other data. 

    "message": "Success: Added new sample <a href="/samples/details/7ed40eac1bbb248678424b2440f6b1f8/">7ed40eac1bbb248678424b2440f6b1f8.</a>"

That message links to only a single one of the samples uploaded from the zip folder. The other samples are left out and nothing is returned about them. It seems this message should be updated to reference all of the uploaded samples.

**Comment by Lambdanaut** [comment]
I will be working on a fix for this that will just return a longer string, separating each sample's message with a `<br />`

Sound good?

**Comment by mgoffin** [comment]
I don't think that is the right solution. Yes we have some messages being returned which have HTML in them simply due to the fact that that's how the handler currently returns values. But we should move away from that as much as possible and leave that to views and other things that are UI-related.

**Comment by Lambdanaut** [comment]
On one hand I agree, but on the other hand if the message value already exists like that and isn't expected to be removed from the API in a future version, editing it should be fine? 

Do you have any alternative suggestions? :)

**Comment by frbapolkosnik** [comment]
Does this happen when you upload it as type RAW or when you upload it as type of 7z/Zip/RAR?

**Comment by Lambdanaut** [comment]
7z/Zip/RAR

**Comment by mgoffin** [comment]
In the API if a list of MD5's are returned when uploading Samples we basically select the first one, generate the other "required" parts of the response (url to the API details of that new Sample), and return.

If all you are looking for are the IDs of the new Samples, we can have an "ids" field which is a list of strings which are the IDs in the rest of the list, or all of the ids and leave "id" empty. I don't agree with the idea of "well it's not how it is now and it'll get fixed later so let's pile on the work until someone does it".

**Comment by Lambdanaut** [comment]
I don't agree with the idea of "well it's not how it is now and it'll get fixed later so let's pile on the work until someone does it".

Fair enough. Just probing :D :+1: 

I think the idea of having a list of ids returned under the "ids" value isn't bad. I'll write a patch for that. 

Thanks Mike.

**Comment by Lambdanaut** [comment]
After looking further, I'm thinking that what would really be useful along with the list of IDs, would be a list of the md5 hashes of the uploaded samples. Reason being, without them you can't construct the URLs to the uploaded samples. There's not much use to the API without being able to see the uploaded files in Crits. 

Thoughts?

**Comment by mgoffin** [comment]
Might as well just construct the URLs for all of them in the API (it's already done for the first one) and put it in a `urls` field.

**Comment by Lambdanaut** [comment]
Yes, but that URL is to the API and uses the sample's ID. The links to the samples in Crits use the MD5 hash to build it.

**Comment by mgoffin** [comment]
Yes. Keep in mind in order to add a sample you either need 1) the MD5, or 2) the data (which you can calculate the MD5. In both cases you already have the MD5 on your end without the need of the API to do it for you. Also in the future when we don't require an MD5, this would cause a problem.
